### PR TITLE
Add Unicorn Prometheus metrics

### PR DESCRIPTION
**Pull request by LS80**
```
ruby_unicorn_active_workers_total
ruby_unicorn_request_backlog_total
ruby_unicorn_workers_total
```

### CODE REVIEW

File: config/unicorn.rb [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
+require 'prometheus_exporter/instrumentation'
+require_relative 'prometheus_client'
+
+port_http = ENV.fetch('PORT_HTTP')
+pid_file = '/tmp/unicorn.pid'
+
+PrometheusExporter::Instrumentation::Unicorn.start(
+  pid_file: pid_file,
+  listener_address: "0.0.0.0:#{port_http}",
+  client: nil # Use default client
```
Comment (sgerrand) [inline config/unicorn.rb]:
```suggestion
```
The implementation in https://github.com/FundingCircle/shipment_tracker/blob/2e48e4e90d18fdb86f2727aae0e27b7d06966f62/config/initializers/delayed_job.rb#L5 didn't pass a `nil` `client:` in. Should that aspect of the configuration be the same?

Comment (LS80) [reply config/unicorn.rb]: Yes it shouldn't be made a required keyword argument if it actually has a default
https://github.com/discourse/prometheus_exporter/blob/fc8758cb6c749e281d893620483306477a7e4e70/lib/prometheus_exporter/instrumentation/unicorn.rb#L12.

**Comment by LS80** [comment]
Yes indeed that's why I haven't merged it yet, as it's misleading. I'll probably have to change the Unicorn instrumentation to be able to add a unique label per web process, so that the values can be aggregated in Prometheus. I'll try to make a change upstream instead of monkey patching it.

**Comment by bliof** [comment]
The code of the gem mentions `hostname` and some have `pid` but most probably the easiest is to specify both manually https://github.com/discourse/prometheus_exporter#metrics-default-prefix--labels

**Comment by LS80** [comment]
That won't work because we aren't using the single process mode https://github.com/discourse/prometheus_exporter/pull/20.

I've opened https://github.com/discourse/prometheus_exporter/pull/90.

File: config/unicorn.rb (hunk 1 above)
Comment (LS80) [reply config/unicorn.rb]: I've opened https://github.com/discourse/prometheus_exporter/pull/90.

Comment (LS80) [reply config/unicorn.rb]: discourse/prometheus_exporter#90 has been merged. I've bumped the gem to use the master branch until there's a new release.

We now have a hostname label for the metrics.
```
# HELP ruby_unicorn_workers_total Number of unicorn workers.
# TYPE ruby_unicorn_workers_total gauge
ruby_unicorn_workers_total{hostname="e2cf40e4314e"} 4
ruby_unicorn_workers_total{hostname="ca61920edc58"} 4
```

**Comment by LS80** [comment]
discourse/prometheus_exporter#90 has been merged. I've bumped the gem to use the master branch until there's a new release.

We now have a hostname label for the metrics.
```
# HELP ruby_unicorn_workers_total Number of unicorn workers.
# TYPE ruby_unicorn_workers_total gauge
ruby_unicorn_workers_total{hostname="e2cf40e4314e"} 4
ruby_unicorn_workers_total{hostname="ca61920edc58"} 4
```

@bliof

**Comment by LS80** [comment]
The `git-worker` is crashing so this is blocked for now. I only see a huge core dump in Graylog so no idea what the problem is yet.

**Comment by LS80** [comment]
Problem fixed by bumping the library versions in the Docker image.

**Pull request merged**